src/main/java/seedu/address/model/tuitionclass/UniqueClassList.java

```diff
@@ -165,6 +165,12 @@ public void setClass(TuitionClass target, TuitionClass editedClass) {
         if (!target.isSameClass(editedClass) && contains(editedClass)) {
             throw new DuplicateClassException();
         }
+        //check for overlapping ClassTiming.
+        for (TuitionClass tuitionClass : internalList) {
```

Comment:
the logic for checking overlap is already written in isValidTiming. Using condition 
```
if (!target.isSameClass(editedClass) && !isValidTiming(editedClass)) 
```
would suffice

Reply:
ok thanks